diff --git a/devel_generate/taxonomy.devel_generate.inc b/devel_generate/taxonomy.devel_generate.inc
index 70c4162..2d1fc45 100644
--- a/devel_generate/taxonomy.devel_generate.inc
+++ b/devel_generate/taxonomy.devel_generate.inc
@@ -15,22 +15,19 @@ function _taxonomy_devel_generate($object, $field, $instance, $bundle) {
   // will not result in any tags being added.
   $machine_name = $field['settings']['allowed_values'][0]['vocabulary'];
   $vocabulary = taxonomy_vocabulary_machine_name_load($machine_name);
-  if ($max = db_query('SELECT MAX(tid) FROM {taxonomy_term_data} WHERE vid = :vid', array(':vid' => $vocabulary->vid))->fetchField()) {
-    $candidate = mt_rand(1, $max);
-    $query = db_select('taxonomy_term_data', 't');
-    $tid = $query
+  $query = db_select('taxonomy_term_data', 't');
+  $tid = $query
               ->fields('t', array('tid'))
               ->condition('t.vid', $vocabulary->vid, '=')
-              ->condition('t.tid', $candidate, '>=')
+              ->orderRandom()
               ->range(0,1)
               ->execute()
               ->fetchField();
-    // If there are no terms for the taxonomy, the query will fail, in which
-    // case we return NULL.
-    if ($tid === FALSE) {
-      return NULL;
-    }
-    $object_field['tid'] = (int) $tid;
-    return $object_field;
+  // If there are no terms for the taxonomy, the query will fail, in which
+  // case we return NULL.
+  if ($tid === FALSE) {
+    return NULL;
   }
+  $object_field['tid'] = (int) $tid;
+  return $object_field;
 }
